src/main/java/com/linecorp/armeria/client/http/HttpRequestSubscriber.java

```diff
@@ -135,8 +139,23 @@ private void writeFirstHeader() {
                          host, firstHeaders.method().name(), firstHeaders.path());
         logBuilder.attr(RequestLog.HTTP_HEADERS).set(firstHeaders);
 
-        encoder.writeHeaders(ctx, id, streamId(), firstHeaders, false).addListener(this);
-        ctx.flush();
+        switch (firstHeaders.method()) {
+            case GET:
+            case HEAD:
+            case PATCH:
+            case DELETE:
+            case TRACE:
+            case CONNECT:
+                // Do not send any content for the methods which is not supposed to have content.
+                // To be precise, the RFC 2616 does not prohibit sending non-empty content for these methods,
+                // but it is discouraged to do so.
+                // See: http://stackoverflow.com/q/978061
```

Comment:
From a quick reading of the spec, using a CONTINUATION frame might help. It means that for requests with no content, we would always send an empty continuation frame with end headers flag set. The performance implications of sending another packet would probably be significant though...

I'd like to solve this without breaking compliance (admittedly it's ElasticSearch experience that has me concerned in the first place) but am OK with discussing more in an issue and rethinking.